x/deployment/keeper/key.go

```diff
@@ -18,16 +18,22 @@ var (
 func deploymentKey(id types.DeploymentID) []byte {
 	buf := bytes.NewBuffer(deploymentPrefix)
 	buf.Write(id.Owner.Bytes())
-	binary.Write(buf, binary.BigEndian, id.DSeq)
+	if err := binary.Write(buf, binary.BigEndian, id.DSeq); err != nil {
+		panic(err)
+	}
 	return buf.Bytes()
 }
 
 // groupKey provides prefixed key for a Group's marshalled data.
 func groupKey(id types.GroupID) []byte {
 	buf := bytes.NewBuffer(groupPrefix)
 	buf.Write(id.Owner.Bytes())
-	binary.Write(buf, binary.BigEndian, id.DSeq)
-	binary.Write(buf, binary.BigEndian, id.GSeq)
+	if err := binary.Write(buf, binary.BigEndian, id.DSeq); err != nil {
+		panic(err)
```

Comment:
generally we would want to avoid using panics as they immediately shutdown app.
maybe this way?
```go
func groupKey(id types.GroupID) ([]byte, error)
```

Reply:
In the keepers we have a pattern of panic-ing on unexpected behavior. Safety over liveness as a design decision.